### PR TITLE
Logic Integration: AnalyticsService with calorie math and week bucketing

### LINKED ISSUE #101: Logic Integration: AnalyticsService - Calorie math and mapping
﻿### Task
Move analytics logic from SqlWorkoutAnalyticsStore.cs to service layer.

### Requirements
* Implementation: Perform the proportional calorie math ((totalCalories * setCount / totalSets)) in the AnalyticsService.
* Mapping: Map the raw WorkoutLog entity to the WorkoutSessionDetail DTO within this service.

### LINKED ISSUE #102: Logic Integration: AnalyticsService - Date math and week bucketing
﻿### Task
Move date logic from SqlWorkoutAnalyticsStore.cs to service layer.

### Requirements
* Implementation: Move date math for week buckets (GetMondayOfWeek, 4-week loops) into the AnalyticsService.

### PR DESCRIPTION
**Pull request by AndaChiorean**
## Summary
- Ported ExerciseCalorieCalculator from VibeCoders into ClassLibrary/Services with the MET lookup table and standard metabolic formula (MET x weight x hours)
- Created AnalyticsService that centralizes workout analytics previously scattered across VibeCoders repositories:
  - Calorie computation per exercise with automatic MET lookup for unknown exercises
  - Intensity tagging based on average MET (light < 3.0, moderate < 6.0, intense >= 6.0)
  - 4-week consistency bucketing with Monday-aligned weeks for the dashboard view
  - Dashboard summary with total workouts, active time over 7 days, and preferred workout detection
- Added 27 unit tests covering MET lookup, calorie formula math, intensity boundary conditions, week alignment, and bucket consistency

Closes #101, closes #102

### CODE REVIEW

File: Tests/AnalyticsServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    {
+        var service = this.CreateService();
+        Assert.Equal("intense", service.DetermineIntensityTag(6.0f));
+    }
+
+    [Fact]
+    public async Task GetConsistencyLastFourWeeks_ReturnsExactlyFourBuckets()
+    {
+        this.analyticsRepo
+            .Setup(r => r.GetWorkoutsInRangeAsync(It.IsAny<int>(), It.IsAny<DateTime>(), It.IsAny<DateTime>()))
```
Comment (MarioMaxim18) [inline Tests/AnalyticsServiceTests.cs]: abbreviation

**Review by MarioMaxim18 — CHANGES_REQUESTED** [review]

File: Tests/AnalyticsServiceTests.cs (hunk 1 above)
Comment (AndaChiorean) [reply Tests/AnalyticsServiceTests.cs]: Fixed, expanded to full parameter name.

**Review by MarioMaxim18 — APPROVED** [review]

**Comment by MarioMaxim18** [comment]
solve the conflict

**Comment by MarioMaxim18** [comment]
@copilot resolve the merge conflicts in this pull request
